core/src/main/java/com/redhat/lightblue/client/Query.java

```diff
@@ -0,0 +1,245 @@
+package com.redhat.lightblue.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ContainerNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * A query expression
+ */
+public class Query extends Expression
+    implements Update.UpdateQuery {
+
+    public static final BinOp eq=BinOp.eq;
+    public static final BinOp neq=BinOp.neq;
+    public static final BinOp lt=BinOp.lt;
+    public static final BinOp gt=BinOp.gt;
+    public static final BinOp lte=BinOp.lte;
+    public static final BinOp gte=BinOp.gte;
+
+    public static final NaryOp in=NaryOp.in;
+    public static final NaryOp nin=NaryOp.nin;
+
+    public static final ArrOp any=ArrOp.any;
+    public static final ArrOp all=ArrOp.all;
+    public static final ArrOp non=ArrOp.none;
+
+    public static final LogOp and=LogOp.and;
+    public static final LogOp or=LogOp.or;
```

Comment:
Operators are now inner classes, so it is now much easier to write
Query.eq, instead of Query.BinOp.eq. Also, this way you don't need to know
a BinOp or NaryOp exists, all you need to know is eq. Simply importing
Query will bring in all the names necessary to build all queries.

On Mon, Aug 10, 2015 at 1:58 PM, Dennis Crissman <notifications@github.com>
wrote:

> In core/src/main/java/com/redhat/lightblue/client/Query.java
> <https://github.com/lightblue-platform/lightblue-client/pull/144#discussion_r36676775>
> :
>
> > +    public static final BinOp eq=BinOp.eq;
> > +    public static final BinOp neq=BinOp.neq;
> > +    public static final BinOp lt=BinOp.lt;
> > +    public static final BinOp gt=BinOp.gt;
> > +    public static final BinOp lte=BinOp.lte;
> > +    public static final BinOp gte=BinOp.gte;
> > +
> > +    public static final NaryOp in=NaryOp.in;
> > +    public static final NaryOp nin=NaryOp.nin;
> > +
> > +    public static final ArrOp any=ArrOp.any;
> > +    public static final ArrOp all=ArrOp.all;
> > +    public static final ArrOp non=ArrOp.none;
> > +
> > +    public static final LogOp and=LogOp.and;
> > +    public static final LogOp or=LogOp.or;
>
> I still think it is unnecessary, but not enough to block the PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/144/files#r36676775>
> .
>